auth-web/src/components/pay/eft/ShortNameLinkingDialog.vue

```diff
@@ -0,0 +1,236 @@
+import { ShortNameResponseStatus } from '@/util/constants'
+
+interface ShortNameLinkingDialog {
+  shortNameLookupKey: number
+  selectedShortName: object
+  selectedAccount: object
+  accountLinkingErrorDialogTitle: string
+  accountLinkingErrorDialogText: string
+}
+
+export default defineComponent({
+  name: 'UnlinkedShortNameTable',
+  components: { ModalDialog, ShortNameLookup },
+  props: {
+    isShortNameLinkingDialogOpen: {
+      type: Boolean,
+      default: false
+    },
+    selectedShortName: {
+      default: {}
+    }
+  },
+  emits: ['on-link-account', 'close-short-name-linking-dialog'],
+  setup (props, { emit }) {
+    const accountLinkingDialog: Ref<InstanceType<typeof ModalDialog>> = ref(null)
+    const accountLinkingErrorDialog: Ref<InstanceType<typeof ModalDialog>> = ref(null)
+    const state = reactive<ShortNameLinkingDialog>({
+      shortNameLookupKey: 0,
+      selectedShortName: {},
+      selectedAccount: {},
+      accountLinkingErrorDialogTitle: '',
+      accountLinkingErrorDialogText: ''
+    })
+
+    function openAccountLinkingDialog (item: EFTShortnameResponse) {
+      state.selectedShortName = item
+      accountLinkingDialog.value.open()
+    }
+
+    function resetAccountLinkingDialog () {
+      state.selectedAccount = {}
+      state.shortNameLookupKey++
+      emit('close-short-name-linking-dialog')
+    }
+
+    function cancelAndResetAccountLinkingDialog () {
+      accountLinkingDialog.value.close()
+      resetAccountLinkingDialog()
+    }
+
+    function closeAccountAlreadyLinkedDialog () {
+      accountLinkingErrorDialog.value.close()
+    }
+
+    async function linkAccount () {
+      if (!state.selectedShortName?.id || !state.selectedAccount?.accountId) {
+        return
+      }
+      try {
+        const response = await PaymentService.patchEFTShortname(state.selectedShortName.id, state.selectedAccount.accountId)
+        if (response?.data) {
+          emit('on-link-account', response.data)
+          cancelAndResetAccountLinkingDialog()
+        }
+      } catch (error) {
+        if (error.response.data.type === ShortNameResponseStatus.EFT_SHORT_NAME_ALREADY_MAPPED) {
+          state.accountLinkingErrorDialogTitle = 'Account Already Linked'
+          state.accountLinkingErrorDialogText = 'The selected bank short name is already linked to an account.'
+          cancelAndResetAccountLinkingDialog()
+          accountLinkingErrorDialog.value.open()
+        } else {
+          state.accountLinkingErrorDialogTitle = 'Something Went Wrong'
+          state.accountLinkingErrorDialogText = 'An error occurred while linking the bank short name to an account.'
+          cancelAndResetAccountLinkingDialog()
+          accountLinkingErrorDialog.value.open()
+        }
+        console.error('Failed to patchEFTShortname.', error)
+      }
+    }
+
+    onMounted(async () => {
+    })
+
+    watch(() => [props.selectedShortName, props.isShortNameLinkingDialogOpen],
```

Comment:
You're watching on two values here, which would change the value of `selectedShortNameNewValue`?

Wouldn't the change of isShortNameLinkingDialogOpen be boolean?

which would be passed into `selectedShortNameNewValue`?

Reply:
Probably better to use `props.selectedShortName` instead of `selectedShortNameNewValue` (which can be triggered by the dialog opening)